storage/spanner.go

```diff
@@ -0,0 +1,984 @@
+	if !ok {
+		return errors.New("ResourceNotFoundException", table)
+	}
+	pKey := tableConf.PartitionKey
+	var pValue interface{}
+	var sValue interface{}
+	sKey := tableConf.SortKey
+
+	cols := []string{}
+	var key spanner.Key
+	var m1 = make(map[string]interface{})
+
+	for k, v := range m {
+		m1[k] = v
+		if k == pKey {
+			pValue = v
+			delete(m, k)
+			continue
+		}
+		if k == sKey {
+			delete(m, k)
+			sValue = v
+			continue
+		}
+		cols = append(cols, k)
+	}
+	if sValue != nil {
+		key = spanner.Key{pValue, sValue}
+	} else {
+		key = spanner.Key{pValue}
+	}
+	_, err = s.getSpannerClient(table).ReadWriteTransaction(ctx, func(ctx context.Context, t *spanner.ReadWriteTransaction) error {
+		tmpMap := map[string]interface{}{}
+		for k, v := range m {
+			tmpMap[k] = v
+		}
+		if len(eval.Attributes) > 0 || expr != nil {
+			status, _ := evaluateConditionalExpression(ctx, t, table, m1, eval, expr)
+			if !status {
+				return errors.New("ConditionalCheckFailedException")
+			}
+		}
+		table = changeTableNameForSP(table)
+
+		r, err := t.ReadRow(ctx, table, key, cols)
+		if err != nil {
+			return errors.New("ResourceNotFoundException", err)
+		}
+		rs, err := parseRowForNull(r, colDLL, cols)
+		if err != nil {
+			return err
+		}
+		for k, v := range tmpMap {
+			v1, ok := rs[k]
+			if ok {
+				switch v1.(type) {
+				case []interface{}:
+					var ifaces1 []interface{}
+					ba, ok := v.([]byte)
+					if ok {
+
+						json.Unmarshal(ba, &ifaces1)
+					} else {
+						ifaces1 = v.([]interface{})
+					}
+					m1 := map[interface{}]struct{}{}
+					ifaces := v1.([]interface{})
+					for i := 0; i < len(ifaces); i++ {
+						m1[reflect.ValueOf(ifaces[i]).Interface()] = struct{}{}
+					}
+					for i := 0; i < len(ifaces1); i++ {
+
+						delete(m1, reflect.ValueOf(ifaces1[i]).Interface())
+					}
+					ifaces = []interface{}{}
+					for k := range m1 {
+						ifaces = append(ifaces, k)
+					}
+					tmpMap[k] = ifaces
+				default:
+					logger.LogDebug(reflect.TypeOf(v).String())
+				}
+			}
+		}
+		tmpMap[pKey] = pValue
+		if sValue != nil {
+			tmpMap[sKey] = sValue
+		}
+		ddl := models.TableDDL[table]
+
+		for k, v := range tmpMap {
+			t, ok := ddl[k]
+			if t == "BYTES(MAX)" && ok {
+				ba, err := json.Marshal(v)
+				if err != nil {
+					return errors.New("ValidationException", err)
+				}
+				tmpMap[k] = ba
+			}
+		}
+		mutation := spanner.InsertOrUpdateMap(table, tmpMap)
+		err = t.BufferWrite([]*spanner.Mutation{mutation})
+		if err != nil {
+			return errors.New("ResourceNotFoundException", err)
+		}
+		return nil
+	})
+	return err
+}
+
+// SpannerRemove - Spanner Remove functionality like update attribute
+func (s Storage) SpannerRemove(ctx context.Context, table string, m map[string]interface{}, eval *models.Eval, expr *models.UpdateExpressionCondition, colsToRemove []string) error {
+
+	_, err := s.getSpannerClient(table).ReadWriteTransaction(ctx, func(ctx context.Context, t *spanner.ReadWriteTransaction) error {
```

Comment:
No need to use a `ReadWriteTransaction`, just call `Client.Apply`.

Reply:
@rosspatil